Sources/MessagingPush/MessagingPush+AppDelegate.swift

```diff
@@ -64,7 +64,11 @@ extension MessagingPushImplementation {
         switch response.actionIdentifier {
         case UNNotificationDefaultActionIdentifier: // push notification was touched.
             if let deepLinkUrl = pushContent.deepLink {
-                deepLinkUtil.handleDeepLink(deepLinkUrl)
+                // A hack to get an instance of deepLinkUtil without making it a property of the MessagingPushImplementation class. deepLinkUtil is not available to app extensions but MessagingPushImplementation is.
+                // We get around this by getting a instance in this function, only.
+                if let deepLinkUtil = sdkInitializedUtil.postInitializedData?.diGraph.deepLinkUtil {
```

Comment:
Is there anything wrong with `deepLinkUtil` as a property of `MessagingPushImplementation`? because we have `digraph` available in there and we can initialize it easily?

Reply:
No, with the way the SDK project is currently setup, we cannot add `deepLinkUtil` as a property of `MessagingPushImplementation`. I tried to explain this in the comments above the `if let ....` statement here. I would love to improve the comment if it doesn't explain it well enough.

Reply:
Yeah, the comment for me, explains `deepLinkUtil ` is not available in `app extensions`  but we have a workaround to make it available.

But it doesn't explain why `deepLinkUtil` could not be added to `MessagingPushImplementation` as a property and we needed a workaround, which you explain can't happen because the project isn't setup that way. 

Can we please create a ticket to track this? I believe `DeepLinkUtil` should be part of `digraph` and if the instance is available in `MessagingPushImplementation`, we should have been able to initialize it?

Reply:
Yes I will be making a ticket for this.